pkg/render/common/components/components_test.go

```diff
@@ -93,6 +96,111 @@ var _ = Describe("Common components render tests", func() {
+							unhandledFields = append(unhandledFields, fullName)
+						}
+					}
+				}
+
+			}
+			findUnhandled("", reflect.TypeOf(overrides))
+			if expectUnhandled {
+				Expect(unhandledFields).NotTo(BeEmpty())
+			} else {
+				Expect(unhandledFields).To(BeEmpty())
+			}
+		},
+		Entry("APIServerDeployment", &v1.APIServerDeployment{}, false),
+		Entry("CalicoKubeControllersDeployment", &v1.CalicoKubeControllersDeployment{}, false),
+		Entry("CalicoNodeDaemonSet", &v1.CalicoNodeDaemonSet{}, false),
+		Entry("CalicoNodeWindowsDaemonSet", &v1.CalicoNodeWindowsDaemonSet{}, false),
+		Entry("CalicoWindowsUpgradeDaemonSet", &v1.CalicoWindowsUpgradeDaemonSet{}, false),
+		Entry("ComplianceBenchmarkerDaemonSet", &v1.ComplianceBenchmarkerDaemonSet{}, false),
+		Entry("ComplianceControllerDeployment", &v1.ComplianceControllerDeployment{}, false),
+		Entry("ComplianceReporterPodTemplate", &v1.ComplianceReporterPodTemplate{}, false),
+		Entry("ComplianceServerDeployment", &v1.ComplianceServerDeployment{}, false),
+		Entry("ComplianceSnapshotterDeployment", &v1.ComplianceSnapshotterDeployment{}, false),
+		Entry("CSINodeDriverDaemonSet", &v1.CSINodeDriverDaemonSet{}, false),
+		Entry("DashboardsJob", &v1.DashboardsJob{}, false),
+		Entry("DexDeployment", &v1.DexDeployment{}, false),
+		Entry("ECKOperatorStatefulSet", &v1.ECKOperatorStatefulSet{}, false),
+		// EgressGateway operates as a top-level CR and also as its own customization
+		// structure, so it does have fields other than those covered by the override
+		// machinery.
+		Entry("EgressGateway", &v1.EgressGateway{}, false,
+			"TypeMeta",
+			"ObjectMeta",
+			"Spec.Replicas",
+			"Spec.IPPools",
+			"Spec.ExternalNetworks",
+			"Spec.LogSeverity",
+			"Spec.EgressGatewayFailureDetection",
+			"Spec.AWS",
+			"Status",
+		),
+		Entry("EKSLogForwarderDeployment", &v1.EKSLogForwarderDeployment{}, false),
+		Entry("ElasticsearchMetricsDeployment", &v1.ElasticsearchMetricsDeployment{}, false),
+		Entry("FluentdDaemonSet", &v1.FluentdDaemonSet{}, false),
+		Entry("GuardianDeployment", &v1.GuardianDeployment{}, false),
+		Entry("IntrusionDetectionControllerDeployment", &v1.IntrusionDetectionControllerDeployment{}, false),
+		Entry("Kibana", &v1.Kibana{}, false),
+		Entry("L7LogCollectorDaemonSet", &v1.L7LogCollectorDaemonSet{}, false),
+		Entry("LinseedDeployment", &v1.LinseedDeployment{}, false),
+		Entry("ManagerDeployment", &v1.ManagerDeployment{}, false),
+		Entry("PacketCaptureAPIDeployment", &v1.PacketCaptureAPIDeployment{}, false),
+		Entry("PolicyRecommendationDeployment", &v1.PolicyRecommendationDeployment{}, false),
+		Entry("TyphaDeployment", &v1.TyphaDeployment{}, false),
+
+		// This last entry checks that the code above really does identify when a
+		// structure has unhandled fields.  To do this we can use any available structure
+		// with some field names that don't overlap with those that are handled by our
+		// override code.
+		Entry("Installation", &v1.Installation{}, true),
```

Comment:
Is this a test that we need to keep, i.e. a test that the test is doing what it should? 

I would personally be fine with you testing this once manually to ensure it. If that's not quite enough for you, then would you be willing to pull the findUnhandled function out of the test and put it at a higher level and write a separate block for this? We'll never want expectUnhandled to be true for any case other than testing the test function.

Reply:
I'm ok with moving `findUnhandled` to a higher level - although even that is a touch fiddly because it needs changes to how `unhandledFields` is collected.  I wouldn't like to remove the Installation case or make separate blocks for the positive and negative tests - either of those facilitates someone accidentally breaking the test and then not realizing.

Reply:
Would you ever need more than one test for the negative case? Isn't this really just a test for the `findUnhandledFunction` to ensure it's working correctly, as opposed to testing that the overrides only have the supported fields?

If this is the case, it seems like this is merging two somewhat disjointed test cases into one.

Reply:
Maybe I'm not following your point here, but for me it's essential for the positive and negative cases to be using the same code.  I think I'd need to see a precise proposed code change to understand precisely.  Anyway I'll merge this PR now, but happy to discuss possible improvements further if you feel like proposing those.